### DISCUSSION TITLE
How to check if component is being server side rendered

### DISCUSSION THREAD

**Discussion by serendipity1004** [vercel/next.js#14617] [Help]
I am currently working on a project that uses Apollo GraphQL and NextJS. I have structured the project quite commonly by having components folder and containers folder where container folders include Apollo states. At first, I thought it would be possible to call `getServerSideProps` from the components and containers but it doesn't seem to be the case. So currently I am calling the queries in the page that uses the containers and using `useQuery` hook in the container to retrieve the state. Then I begin to wonder if the container will be server rendered or not. How do I check if a component is being fully server rendered with the states that I expect to be present?

**Comment by IamLizu** [comment]
> I thought it would be possible to call `getServerSideProps` from the components and containers but it doesn't seem to be the case.

Yes, `getServerSideProps` should be used in Pages only.


> Then I begin to wonder if the container will be server rendered or not.

When you use `getServerSideProps`, your html is generated on each request.

Please take a look at the [docs](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering)  for detailed information.
